### ISSUE TITLE
Invalid count argument

### ISSUE THREAD

**Issue by grdnrio** [cloudposse/terraform-aws-iam-s3-user#12]
Hey, I'm using 0.4.0 to create an temp IAM user to access a pre-existing bucket in S3. Running Terraform 0.12.18 (latest).

Currently getting this error.

```
Error: Invalid count argument

  on .terraform/modules/s3_user.s3_user.label/main.tf line 72, in data "null_data_source" "tags_as_list_of_maps":
  72:   count = local.enabled ? length(keys(local.tags)) : 0

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```
Code snippet being referenced.

```
data "null_data_source" "tags_as_list_of_maps" {
  count = local.enabled ? length(keys(local.tags)) : 0

  inputs = merge(
    {
      "key"   = keys(local.tags)[count.index]
      "value" = values(local.tags)[count.index]
    },
    var.additional_tag_map
  )
}
```
Seems that [#4149](https://github.com/hashicorp/terraform/issues/4149) will resolve this with a secondary plan stage, but which `-target` argument should I be using in the meantime?

**Comment by nitrocode** [comment]
@grdnrio are you still hitting this issue?

**Comment by Nuru** [comment]
This was fixed in v0.5.0
